File: Java/Begineer Problem/Logic Building (Patterns)/pattern12.java
import java.util.Scanner;

public class pattern12 {
    public static void pattern12(int n) {
        for(int i = 1; i <= n; i++){
            for(int j = 1; j <= i; j++){
                System.out.print(j);
            }
            for(int space = 1; space <= 2 * n - 2 * i; space++){
                System.out.print(" ");
            }
            for (int j = i; j >= 1; j--) {
                System.out.print(j);
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int testCases = sc.nextInt();
        while(testCases-- > 0){
            int n = sc.nextInt();
            pattern12(n);
        }
    }   
}

File: C++/Begineer Problem/Logic Building (Patterns)/pattern12.cpp
#include<bits/stdc++.h>
using namespace std;

void pattern12(int n) {
    for(int i = 1; i <= n; i++){
        for(int j = 1; j <= i; j++){
            cout << j;
        }
        for(int space = 1; space <= 2 * n - 2 * i; space++){
            cout << " ";
        }
        for (int j = i; j >= 1; j--) {
            cout << j;
        }
        cout << endl;
    }
}
int main()
{
    int testCases;
    cin >> testCases;
    while (testCases--){
        int n;
        cin >> n;
        pattern12(n);
    }
    return 0;
}